src/realm/sync/client.cpp

```diff
@@ -510,22 +519,25 @@ void ClientImpl::shutdown() noexcept
 
 void ClientImpl::register_unactualized_session_wrapper(SessionWrapper* wrapper, ServerEndpoint endpoint)
 {
+    bool retrigger;
     // Thread safety required.
-
-    std::lock_guard lock{m_mutex};
-    REALM_ASSERT(m_actualize_and_finalize);
-    m_unactualized_session_wrappers.emplace(wrapper, std::move(endpoint)); // Throws
-    bool retrigger = !m_actualize_and_finalize_needed;
-    m_actualize_and_finalize_needed = true;
-    // The conditional triggering needs to happen before releasing the mutex,
-    // because if two threads call register_unactualized_session_wrapper()
-    // roughly concurrently, then only the first one is guaranteed to be asked
-    // to retrigger, but that retriggering must have happened before the other
-    // thread returns from register_unactualized_session_wrapper().
-    //
-    // Note that a similar argument applies when two threads call
-    // register_abandoned_session_wrapper(), and when one thread calls one of
-    // them and another thread call the other.
+    {
```

Comment:
This just moves the lock_guard into its own scope, since calling `m_actualize_and_finalize->trigger()` wakes up the event loop thread which will immediately try to acquire `m_mutex` and wait on a cond var - so this should hopefully reduce the likelihood the event loop will have to block for the mutex.